File: leetcode/3996.py
class Solution:
    def canReach(self, start: list[int], target: list[int]) -> bool:
        def isBlack(pos):
            row, col = pos
            return (row + col) % 2 # is sum even?
        # start and end on same color?
        return isBlack(start) == isBlack(target)

File: leetcode/3996.kt
class Solution {
    fun canReach(start: IntArray, target: IntArray): Boolean {
        fun isBlack(pos: IntArray): Boolean {
            val (row, col) = pos
            return (row + col) % 2 == 0
        }
        return isBlack(start) == isBlack(target)
    }
}

/*
class Solution {
    fun canReach(start: IntArray, target: IntArray): Boolean {
        fun isBlack(pos: IntArray): Boolean {
            val row: Int = pos[0]
            val col: Int = pos[1]
            return (row + col) % 2 == 0
        }
        return isBlack(start) == isBlack(target)
    }
}
 */
